scanpy/preprocessing/_highly_variable_genes.py

```diff
@@ -175,6 +176,196 @@ def _highly_variable_genes_seurat_v3(
         return df
 
 
+def _highly_variable_pearson_residuals(
+    adata: AnnData,
+    layer: Optional[str] = None,
+    n_top_genes: int = 2000,
+    batch_key: Optional[str] = None,
+    theta: float = 100,
+    clip: Union[Literal['auto', 'none'], float] = 'auto',
+    chunksize: int = 100,
+    subset: bool = False,
+    inplace: bool = True,
+) -> Optional[pd.DataFrame]:
+    """\
+    See `highly_variable_genes`.
+
+    Returns
+    -------
+    Depending on `inplace` returns calculated metrics (:class:`~pd.DataFrame`)
+    or updates `.var` with the following fields
+
+    highly_variable
+        boolean indicator of highly-variable genes
+    means
+        means per gene
+    variances
+        variances per gene 
+    residual_variances
+        Pearson residual variance per gene. Averaged in the case of multiple
+        batches.
```

Comment:
this is something we are not very consistent in scanpy but we should strive to be more so. Can you please end each statement with a dot?

Reply:
see `highly_variable_genes` main function for reference

Reply:
ok! fixed that throughout the files I am working on